lib/temporal/activity/context.rb

```diff
@@ -1,13 +1,14 @@
 # This context class is available in the activity implementation
 # and provides context and methods for interacting with Temporal
 #
+require 'temporal/errors'
 require 'temporal/uuid'
 require 'temporal/activity/async_token'
 
 module Temporal
   class Activity
     class Context
-      def initialize(connection, metadata, config, heartbeat_thread_pool)
+      def initialize(connection, metadata, config, heartbeat_thread_pool, is_shutting_down)
```

Comment:
nit: `is_shutting_down` -> `shutting_down_proc`? Otherwise it feels like a boolean is being passed